proof/invariant-abstract/RISCV64/ArchFinalise_AI.thy

```diff
@@ -959,11 +959,12 @@ lemma reply_unlink_sc_not_live:
   "\<lbrace>obj_at (\<lambda>ko. \<exists>r. ko = Reply r \<and> reply_tcb r = None) reply and invs\<rbrace>
      reply_unlink_sc sc_ptr reply
    \<lbrace>\<lambda>rv. obj_at (\<lambda>ko. \<not> live ko \<and> is_reply ko) reply\<rbrace>"
+  unfolding reply_unlink_sc_def
   apply (wpsimp wp: update_sched_context_obj_at_impossible
                     simple_obj_set_prop_at get_simple_ko_wp
                     set_simple_ko_obj_at_disjoint hoare_vcg_all_lift
-              simp: reply_unlink_sc_def is_reply update_sk_obj_ref_def
-        | wp (once) hoare_drop_imps)+
+              simp:  is_reply
+        | wp (once) hoare_drop_imps | simp)+
```

Comment:
This extra `simp` here seems a bit strange. Do you know why the `wpsimp` isn't enough?

Reply:
I think it is something to do with unification that clarsimp fails to handle.